controllers/operator/apigateway_controller.go

```diff
@@ -203,49 +205,56 @@ func (r *APIGatewayReconciler) reconcileFinalizer(ctx context.Context, apiGatewa
 		controllerutil.AddFinalizer(apiGatewayCR, ApiGatewayFinalizer)
 		if err := r.Client.Update(ctx, apiGatewayCR); err != nil {
 			ctrl.Log.Error(err, "Failed to add API-Gateway CR finalizer")
-			return controllers.ErrorStatus(err, "Could not add API-Gateway CR finalizer")
+			return controllers.ErrorStatus(err, "Could not add API-Gateway CR finalizer", conditions.ReconcileFailed.Condition())
 		}
 	}
 
 	if apiGatewayCR.IsInDeletion() && hasFinalizer(apiGatewayCR) {
 		apiRulesFound, err := apiRulesExist(ctx, r.Client)
 		if err != nil {
-			return controllers.ErrorStatus(err, "Error during listing existing APIRules")
+			return controllers.ErrorStatus(err, "Error during listing existing APIRules", conditions.ReconcileFailed.Condition())
 		}
-		if apiRulesFound {
+		if len(apiRulesFound) > 0 {
 			return controllers.WarningStatus(errors.New("could not delete API-Gateway CR since there are APIRule(s) that block its deletion"),
-				"There are APIRule(s) that block the deletion of API-Gateway CR. Please take a look at kyma-system/api-gateway-controller-manager logs to see more information about the warning")
+				"There are APIRule(s) that block the deletion of API-Gateway CR. Please take a look at kyma-system/api-gateway-controller-manager logs to see more information about the warning",
+				conditions.DeletionBlockedExistingResources.AdditionalMessage(": "+strings.Join(apiRulesFound, ",")).Condition())
 		}
 
 		oryRulesFound, err := oryRulesExist(ctx, r.Client)
 		if err != nil {
-			return controllers.ErrorStatus(err, "Error during listing existing ORY Oathkeeper Rules")
+			return controllers.ErrorStatus(err, "Error during listing existing ORY Oathkeeper Rules", conditions.ReconcileFailed.Condition())
 		}
-		if oryRulesFound {
+		if len(oryRulesFound) > 0 {
 			return controllers.WarningStatus(errors.New("could not delete API-Gateway CR since there are ORY Oathkeeper Rule(s) that block its deletion"),
-				"There are ORY Oathkeeper Rule(s) that block the deletion of API-Gateway CR. Please take a look at kyma-system/api-gateway-controller-manager logs to see more information about the warning")
+				"There are ORY Oathkeeper Rule(s) that block the deletion of API-Gateway CR. Please take a look at kyma-system/api-gateway-controller-manager logs to see more information about the warning",
+				conditions.OathkeeperReconcileFailed.AdditionalMessage(": "+strings.Join(oryRulesFound, ",")).Condition())
```

Comment:
fyi, https://github.com/kyma-project/api-gateway/issues/903

```
with the status Warning we report up to 5 custom resources that are blocking the deletion. The listing format for each is <kind>/<name>.
```

Reply:
I think it won't be meaningful to list only up to 5 resources. I don't think the list you get from apiserver is ordered so you may end up with overriding condition. I would leave listing all conditions and if the logs get cluttered too much, we can adjust it. What do you think?

Reply:
What will your condition message be, when you have 1000 blocking resources?

Reply:
1000 resources! it will generate 1000 log lines though... Maybe we can think about how to handle it differently? Keeping it in condition message seems kind of incorrect to me anyway. If we generate log lines then do we really need it in a condition message?

Reply:
I'd say the Apigateway CR is our main "communication channel" with the user. Having it in the logs only didn't work in the past. We saw that customers were missing this or wondering why it's so complicated. 
Having a shorter list is somehow a compromise in between the two options and we want to see if this is better for the user. Of course we can revisit this, but I think in the first approach we should implement our what we agreed on and check if this is sufficient.